modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/CmgRaftGroupListener.java

```diff
@@ -244,6 +252,23 @@ private void removeNodesFromLogicalTopology(NodesLeaveCommand command) {
         }
     }
 
+    private void changeMetastorageNodes(ChangeMetastorageNodesCommand command) {
+        ClusterState existingState = storageManager.getClusterState();
+
+        assert existingState != null : "Cluster state is not initialized when got " + command;
+
+        ClusterState newState = cmgMessagesFactory.clusterState()
+                .cmgNodes(Set.copyOf(existingState.cmgNodes()))
```

Comment:
Why do we need these `Set.copyOf` here?

Reply:
To make sure we don't get aliasing. This is really cheap and allows to not think about aliasing at all